### PR TITLE
Added the env var VGL_EXCLUDE to list displays to not be interposed.

### PR DESCRIPTION
**Pull request by hernando**
The expected syntax is a semicolon separated list of display names.
e.g.: VGL_EXCLUDE=":0.1;:0.2"

### CODE REVIEW

File: doc/advancedconfig.txt [hunk 1]
```diff
@@ -169,6 +169,23 @@ Each method of configuration takes precedence over the previous method:
...
+	Description :: This option allows the user to provide a list of displays
+	for which VirtualGL must be bypassed. The list is a single string of
+	semicolon separated display names. When a connection to a display
+	server is open, VirtualGL checks if the display name appears in the list
+	(the content of DISPLAY is used in case no name is provided to
+	XOpenDisplay). In that case, the display is appended to the list of
+	displays ignored by VirtualGL. This is useful for multi-GPU machines
+	where a parallel rendering application wants to make use direct use of
+	some of the GPUs and apply VirtualGL interposition only to the GPU used
+	for displayed the final composition.
```
Comment (tribal-tec) [inline doc/advancedconfig.txt]: for displaying...

File: doc/index.html [hunk 2]
```diff
@@ -3312,6 +3312,44 @@ <h2 id="hd0019001">19.1&nbsp;Server Settings</h2>
...
+        This option allows the user to provide a list of displays for which
+        VirtualGL must be bypassed. The list is a single string of semicolon
+        separated display names. When a connection to a display server is open,
+        VirtualGL checks if the display name appears in the list (the content
+        of <code>DISPLAY</code> is used in case no name is provided to
+        <code>XOpenDisplay</code>). In that case, the display is appended to the
+        list of displays ignored by VirtualGL. This is useful for multi-GPU
+        machines where a parallel rendering application wants to make use direct
+        use of some of the GPUs and apply VirtualGL interposition only to the GPU
+        used for displayed the final composition.
```
Comment (tribal-tec) [inline doc/index.html]: Dito

File: doc/advancedconfig.txt [hunk 3]
```diff
@@ -169,6 +169,23 @@ Each method of configuration takes precedence over the previous method:
...
+| Environment Variable | ''VGL_EXCLUDE = ''__''{d[;d...]}''__ |
+| Summary | __''{d[;d...]}''__ = A semicolon separated list of displays to ignore |
+| Image Transports | All |
+| Default Value | None |
+#OPT: hiCol=first
+
+	Description :: This option allows the user to provide a list of displays
+	for which VirtualGL must be bypassed. The list is a single string of
+	semicolon separated display names. When a connection to a display
+	server is open, VirtualGL checks if the display name appears in the list
```
Comment (eile) [inline doc/advancedconfig.txt]: is opened,

File: doc/advancedconfig.txt [hunk 4]
```diff
@@ -169,6 +169,23 @@ Each method of configuration takes precedence over the previous method:
...
+	(the content of DISPLAY is used in case no name is provided to
+	XOpenDisplay). In that case, the display is appended to the list of
+	displays ignored by VirtualGL. This is useful for multi-GPU machines
+	where a parallel rendering application wants to make use direct use of
+	some of the GPUs and apply VirtualGL interposition only to the GPU used
+	for displaying the final composition.
```
Comment (eile) [inline doc/advancedconfig.txt]: s/composition/result|image/?

**Comment by hernando** [comment]
This solution works for eqPly with multi-GPU also. I was a bit confused in the beginning because I was expecting a different behavior from Equalizer, but finally I've understood that there's no need to do anything special, just choose the right port.screens to be be excluded.

**Comment by hernando** [comment]
+2 anyone?

File: server/faker-glx.cpp [hunk 5]
```diff
@@ -1944,53 +2030,43 @@ int glXQueryGLXPbufferSGIX(Display *dpy, GLXPbuffer pbuf, int attribute,
...
 	else return NULL;
 }
 
-
-// Hand off to the 3D X server without modification.
-
 Bool glXQueryVersion(Display *dpy, int *major, int *minor)
 {
+	if(vglfaker::isExcluded(dpy))
+		_glXQueryVersion(dpy, major, minor);
```
Comment (tribal-tec) [inline server/faker-glx.cpp]: missing return

File: server/faker-glx.cpp [hunk 6]
```diff
@@ -1944,53 +2030,43 @@ int glXQueryGLXPbufferSGIX(Display *dpy, GLXPbuffer pbuf, int attribute,
 }
 
 
-// Hand off to the 3D X server without modification.
-
 Bool glXQueryExtension(Display *dpy, int *error_base, int *event_base)
 {
+	if(vglfaker::isExcluded(dpy))
+		_glXQueryExtension(dpy, error_base, event_base);
```
Comment (tribal-tec) [inline server/faker-glx.cpp]: missing return

File: server/faker-glx.cpp [hunk 7]
```diff
@@ -1301,6 +1353,11 @@ int glXGetFBConfigAttribSGIX(Display *dpy, GLXFBConfigSGIX config,
 
 GLXFBConfigSGIX glXGetFBConfigFromVisualSGIX(Display *dpy, XVisualInfo *vis)
 {
+	if(vglfaker::isExcluded(dpy))
+	{
+		_glXGetFBConfigFromVisualSGIX(dpy, vis);
```
Comment (tribal-tec) [inline server/faker-glx.cpp]: missing return

File: server/faker-x11.cpp [hunk 8]
```diff
@@ -450,13 +583,15 @@ Display *XOpenDisplay(_Xconst char* name)
 Bool XQueryExtension(Display *dpy, _Xconst char *name, int *major_opcode,
 	int *first_event, int *first_error)
 {
-	Bool retval=True;
-
-	// Prevent recursion
-	if(is3D(dpy))
-		return _XQueryExtension(dpy, name, major_opcode, first_event, first_error);
+	if(is3D(dpy) || vglfaker::isExcluded(dpy))
```
Comment (tribal-tec) [inline server/faker-x11.cpp]: swap operands to be consistent with the rest

**Comment by tribal-tec** [comment]
+1 otherwise

File: server/faker-glx.cpp (hunk 5 above)
Comment (hernando) [inline server/faker-glx.cpp]: Good catch

**Comment by eile** [comment]
+2

**Comment by bilgili** [comment]
+2

**Pull request merged**